sdk/openai/openai-build.yml

```diff
@@ -0,0 +1,78 @@
+steps:
+  - pwsh: |
+      java -version
+      echo "Java version above"
+      # Define repository details
+      $repoUrl = "https://github.com/openai/openai-java.git"
+      $branchName = "next"
+
+      # Clone the repository and checkout the specified branch
+      Write-Host "Cloning repository: $repoUrl"
+      git clone $repoUrl openai-java
+
+      cd openai-java
+
+      #Write-Host "Checking out branch: $branchName"
+      git checkout $branchName
+    displayName: Clone OpenAI Java repository and checkout branch
+    workingDirectory: $(Agent.TempDirectory)
+
+  - pwsh: |
+      # Define the path to the build.gradle.kts file
+      $buildGradlePath = "$(Agent.TempDirectory)/openai-java/build.gradle.kts"
+
+      # Read the contents of the build.gradle.kts file
+      $buildGradleContent = Get-Content -Path $buildGradlePath
+
+      # Extract the version number from the build.gradle.kts file
+      $versionLine = $buildGradleContent | Select-String -Pattern 'version\s*=\s*\"([^\"]+)\"'
+      if ($versionLine -match 'version\s*=\s*\"([^\"]+)\"') {
+        $version = $matches[1]
+        Write-Host "Current version: $version"
+      } else {
+        Write-Host "Version not found"
+        exit 1
+      }
+
+      # Define the new version with '-SNAPSHOT' appended
+      $newVersion = "$version-SNAPSHOT"
+      Write-Host "New version: $newVersion"
+
+      # Update the version in the build.gradle.kts file
+      $updatedContent = $buildGradleContent -replace 'version\s*=\s*\"([^\"]+)\"', "version = `"$global:newVersion`""
+      Set-Content -Path $buildGradlePath -Value $updatedContent
+     
+      Write-Host "##vso[task.setvariable variable=NewVersion]$newVersion"
+    displayName: Find current version and update to SNAPSHOT
+    workingDirectory: $(Agent.TempDirectory)
+
+  - pwsh: |
```

Comment:
Use the [Gradle build task](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/gradle-v4?view=azure-pipelines) instead as it can manage setting the Java version for you and running the build.

Reply:
I think this is why we were filtering out the Java 8 runs as we weren't setting a Java version here, leading to Gradle compilation failures.

But since the Gradle task will manage setting the Java version based on `jdkVersionOption` it'll make everything work happily (and we can hardcode to `1.11` as we don't need to verify their project can build with Java 17, 21, etc).

Reply:
We probably also need to verify their project can be built with Java 17 and 21 as we add Azure-specific features in their source code.

Reply:
> We probably also need to verify their project can be built with Java 17 and 21 as we add Azure-specific features in their source code.

Forget what I was saying above. Yeah. if Stainless team already release, they should already compile the source code and run tests successfully for non-azure endpoint.

Reply:
- task: JavaToolInstaller@1
    inputs:
      versionSpec: '11'
      jdkArchitectureOption: 'x64'
      jdkSourceOption: 'PreInstalled'